tools/cli/adminCommands.go

```diff
@@ -183,6 +186,99 @@ func describeMutableState(c *cli.Context) *adminservice.DescribeMutableStateResp
 	return resp
 }
 
+// RegisterNamespace register a namespace
+func RegisterNamespace(c *cli.Context) error {
+	namespace := getRequiredGlobalOption(c, FlagNamespace)
+
+	description := c.String(FlagDescription)
+	ownerEmail := c.String(FlagOwnerEmail)
+
+	client := cFactory.FrontendClient(c)
+
+	var err error
+	retention := defaultNamespaceRetention
+	if c.IsSet(FlagRetention) {
+		retention, err = timestamp.ParseDurationDefaultDays(c.String(FlagRetention))
+		if err != nil {
+			return fmt.Errorf("option %s format is invalid: %s", FlagRetention, err)
+		}
+	}
+
+	var isGlobalNamespace bool
+	if c.IsSet(FlagIsGlobalNamespace) {
+		isGlobalNamespace, err = strconv.ParseBool(c.String(FlagIsGlobalNamespace))
+		if err != nil {
+			return fmt.Errorf("option %s format is invalid: %w.", FlagIsGlobalNamespace, err)
+		}
+	}
+
+	namespaceData := map[string]string{}
+	if c.IsSet(FlagNamespaceData) {
+		namespaceDataStr := c.String(FlagNamespaceData)
+		namespaceData, err = parseNamespaceDataKVs(namespaceDataStr)
+		if err != nil {
+			return fmt.Errorf("option %s format is invalid: %s", FlagNamespaceData, err)
+		}
+	}
+	if len(requiredNamespaceDataKeys) > 0 {
+		err = checkRequiredNamespaceDataKVs(namespaceData)
+		if err != nil {
+			return fmt.Errorf("namespace data missed required data: %s", err)
+		}
+	}
+
+	var activeClusterName string
+	if c.IsSet(FlagActiveClusterName) {
+		activeClusterName = c.String(FlagActiveClusterName)
+	}
+
+	var clusters []*replicationpb.ClusterReplicationConfig
+	if c.IsSet(FlagClusters) {
+		clusterStr := c.String(FlagClusters)
```

Comment:
Is this one cluster or multiple clusters?

Reply:
this code needs to be refactored imo, rn this is copy-pasted from old code. As an option we can switch to StringSlice flag option (ie. so people can provide the same flag multiple times `--cluster 1 --cluster 2` as it's done here https://github.com/temporalio/tctl/blob/f7472eba8cd1431e3ccd95a51f9b218866ccced5/cli/workflowCommands.go#L251)

To answer the question: it is single cluster. Then it also uses args for additional cluster names which is quite bad UX